src/qt/transactionrecord.cpp

```diff
@@ -58,8 +58,19 @@ QList<TransactionRecord> TransactionRecord::decomposeTransaction(const CWallet *
     int64_t nNet = nCredit - nDebit;
     size_t wtx_size = wtx.vout.size();
     uint256 hash = wtx.GetHash();
+
     std::map<std::string, std::string> mapValue = wtx.mapValue;
 
+    bool fContractPresent = false;
+    NN::ContractType ContractType;
+
+    if (!wtx.GetContracts().empty())
+    {
+         const auto& contract = wtx.GetContracts().begin();
+         fContractPresent = true;
+         ContractType = contract->m_type.Value();
+    }
+
```

Comment:
Any reason to pull this out still? It looks like it's only being used in the original.

Reply:
Oh wait, I see it used below there.

Reply:
Yep. Used twice...

Reply:
And it is better to do it once per transaction rather than once every output...